lib/datadog/ci/ext/environment.rb

```diff
@@ -213,16 +213,19 @@ def extract_github_actions(env)
           ref = env['GITHUB_REF'] if ref.nil? || ref.empty?
           branch, tag = branch_or_tag(ref)
 
+          pipeline_url = "#{env['GITHUB_SERVER_URL']}/#{env['GITHUB_REPOSITORY']}/actions/runs/#{env['GITHUB_RUN_ID']}"
```

Comment:
i. Are we guaranteed that `GITHUB_SERVER_URL` will be present, or should we have a fallback value for it?

---

ii. Also somewhat related, one thing we don't do in this file at all, but perhaps would be worth considering, is to use `Hash#fetch` instead of `Hash#[]` when we're fetching information we consider to be required.

For instance, in this case:

```ruby
[1] pry(main)> env = {}
=> {}
[2] pry(main)> env['GITHUB_SERVER_URL']
=> nil
[3] pry(main)> env.fetch('GITHUB_SERVER_URL')
KeyError: key not found: "GITHUB_SERVER_URL"
from (pry):3:in `fetch'
```

of course, this very much depends on what we want out of this process. Perhaps not having this failures so far is intentional, because we may prefer to have an incorrect metadata value, rather than a failure (...?).

Reply:
i. It _seems_ so: https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables. This has consistently worked for us in our tests. 

ii. Yeah I'd say it's safer to fallback to `nil` rather than failing. We follow CI's documentation pages for the name of these variables, but there are cases in which it's difficult to tell, for example, is `GITHUB_REF_NAME` defined for both PR and merge triggers? Leaving a key empty is safe: we just don't some a piece of metadata in our UI

Reply:
Note that in some cases, like this one, the key would not be empty, but actually incorrect, e.g. `/actions/runs/` if all of the `GITHUB_...` env vars were missing.

Should we test that they exist before using them for templating?

Reply:
we work under the assumption that some env vars are always there. For example, if `GITHUB_REPOSITORY` is missing, we can't associate the test run to a repository, so most of our UI won't work. Test spans will still be in the event platform, with missing or invalid metadata, but I think that's fine, since we don't have any valid value to fallback to.